includes/Checker/Abstract_Check_Runner.php

```diff
@@ -167,4 +254,40 @@ protected function get_checks_to_run() {
 
 		return array_intersect_key( $all_checks, array_flip( $check_slugs ) );
 	}
+
+	/**
+	 * Creates and returns the Check instance.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return Checks An instance of the Checks class.
+	 *
+	 * @throws Exception Thrown if the plugin slug is invalid.
+	 */
+	protected function get_checks_instance() {
+		if ( isset( $this->checks ) ) {
+			return $this->checks;
+		}
+
+		$plugin_slug     = isset( $this->plugin_slug ) ? $this->plugin_slug : $this->get_plugin_param();
```

Comment:
See above regarding naming, let's use `$plugin` instead of `$plugin_slug` to account for the ambiguity of what this value can contain (it's not necessarily a slug).